custom_components/tahoma/__init__.py

```diff
@@ -76,13 +76,25 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     username = entry.data.get(CONF_USERNAME)
     password = entry.data.get(CONF_PASSWORD)
 
+    client = TahomaClient(username, password)
+
     try:
-        client = TahomaClient(username, password)
         await client.login()
+    except TooManyRequestsException:
+        _LOGGER.error("too_many_requests")
+        await client.close()
+        return False
+    except BadCredentialsException:
+        _LOGGER.error("invalid_auth")
+        await client.close()
+        return False
     except Exception as exception:  # pylint: disable=broad-except
         _LOGGER.exception(exception)
+        await client.close()
         return False
 
+    update_interval = entry.options.get(CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL)
```

Comment:
[Here](https://github.com/iMicknl/ha-tahoma/pull/227/files#diff-7132e279b007f1584ef563c8b28cdd00R84) you set a default value, so you should be able to only do `entry.options[CONF_UPDATE_INTERVAL]`

Reply:
I am still figuring this out. The Option Flow is triggered only when you press `options` now, thus the value can be empty.